src/main/java/reflection/ExtensibleFunctionsLoader.java

```diff
@@ -0,0 +1,56 @@
+package reflection;
+
+import extensible.functions.CalculatorFunction;
+import org.reflections.Reflections;
+
+import java.lang.reflect.Method;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+public class ExtensibleFunctionsLoader {
+    private static final String COULD_NOT_BE_FOUND = " could not be found.";
+    CalculatorFunction calculatorFunction;
+
+    public Set<Class<? extends CalculatorFunction>> loadCalculatorFunctionSubClasses() {
+        Reflections reflections = new Reflections(CalculatorFunction.class);
+        return reflections.getSubTypesOf(CalculatorFunction.class);
+    }
+
+    public Class<? extends CalculatorFunction> getSpecificClass(String className) {
```

Comment:
Vlt. hier noch einmal, dass man alle Funktionen in dem Package bekommt

Reply:
wollen wir das denn? ich hätte gedacht, dass alle Kindklassen von dem Interface reichen

Reply:
ja, das stimmt auch. aber wir können ja nicht wissen, wie die klassen genau heißen und welche das sind. deswegen können wir da nicht von dem klassennamen ausgehen, sondern müssen uns den erst holen